lib/resque/worker.rb

```diff
@@ -861,7 +862,11 @@ def perform_with_fork(job, &block)
 
       begin
         @child = fork do
-          unregister_signal_handlers if term_child
+          if term_child
+            unregister_signal_handlers
+          else
+            @signal_handler.reopen
```

Comment:
are any tests failing if you remove this line here? If not, can you add a new test?